modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java

```diff
@@ -240,6 +306,50 @@ public void testFreeMarkerNotificationPickListObjectFieldTerm()
 					notificationQueueEntry.getBody(), StringPool.COMMA)));
 	}
 
+	@Test
+	public void testFreeMarkerNotificationSiteScope() throws Exception {
+		String body = LocalizationUtil.updateLocalization(
+			LocalizedMapUtil.getLocalizedMap(
+				HashMapBuilder.put(
+					LanguageUtil.getLanguageId(LocaleUtil.US),
+					StringUtil.merge(
+						_freeMarkerTermValuesSiteScope.keySet(),
+						StringPool.COMMA)
+				).build()),
+			null, "Body", LanguageUtil.getLanguageId(LocaleUtil.US));
+
+		_executeNotificationObjectActionOnSiteScope(
+			_addNotificationTemplateOnSiteScope(
+				body, NotificationTemplateConstants.EDITOR_TYPE_FREEMARKER,
+				false,
+				Collections.singletonMap(
+					LocaleUtil.US, user1.getEmailAddress())));
+
+		List<NotificationQueueEntry> notificationQueueEntries =
+			notificationQueueEntryLocalService.getNotificationEntries(
+				NotificationConstants.TYPE_EMAIL,
+				NotificationQueueEntryConstants.STATUS_SENT);
+
+		Assert.assertEquals(
+			notificationQueueEntries.toString(), 1,
+			notificationQueueEntries.size());
+
+		notificationQueueEntry = notificationQueueEntries.get(0);
+
+		assertTermValues(
+			Arrays.asList(
+				_freeMarkerTermValuesSiteScope.values(
+				).stream(
+				).map(
+					Object::toString
+				).toArray(
+					String[]::new
+				)),
```

Comment:
Hey @rijema, currently we're not using streams anymore due to performance improvements, could you please try to use another approach here, for example, I'd suggest you to use `TransformUtil.transformToArray` method